akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/BehaviorTestKit.scala

```diff
@@ -60,6 +62,20 @@ object BehaviorTestKit {
 @ApiMayChange
 abstract class BehaviorTestKit[T] {
 
+  /**
+   * Constructs a message using the provided 'messageFactory' to inject a single-use "reply to"
+   * [[akka.actor.typed.ActorRef]], and sends the constructed message to the behavior, recording any [[Effect]]s.
+   *
+   * The returned [[TestInbox]] allows the message sent to the "reply to" `ActorRef` to be asserted on.
+   */
+  def ask[Res](messageFactory: JFunction[ActorRef[Res], T]): TestInbox[Res]
```

Comment:
One thing is weird with returning a TestInbox here: ask will only ever complete with a single message, or no message, but inbox makes it look like it could return multiple responses over time. Would be better with a specific testkit type denoting a single response, not having `ref` etc?

Reply:
What about hiding the inbox and returning the response value (or throwing if no response)